gym_handeye/handeye.py

```diff
@@ -0,0 +1,310 @@
+import gym
+import logging
+
+from gym.spaces import Discrete
+import random
+
+import gym_handeye.utils.utils as utils
+
+ACTION_LOOKUP = {
+    0: 'N',
+    1: 'E',
+    2: 'S',
+    3: 'W',
+    4: 'G',
+    5: 'R'
+}
+
+
+class HandEye(gym.Env):
+    metadata = {'render.modes': ['human']}
+
+    def __init__(self, grid_size, note_in_hand, test_only_changes):
+        logging.debug('Starting environment HandEye')
+        self.grid_size = grid_size
+        self.note_in_hand = note_in_hand
+        self.test_only_changes = test_only_changes
+
+        self.grip_pos_x = None
+        self.grip_pos_y = None
+        self.block_in_hand = False
+        self.block_pos_x = None
+        self.block_pos_y = None
+
+        self.env_size = self.grid_size * self.grid_size + 1
+
+        self.observation_space = Discrete(self.env_size)  # camera monitors the whole grid
+        self.action_space = Discrete(6)  # N - north, S - south, W - west, E - east, G - grip, R - release
+
+        self.goal_generator_state = 0
+
+        self.observation = ['w' for x in range(self.env_size - 1)]  # array with what we observe
+        self.observation.append('0')
```

Comment:
Z tego co pamiętam ostatnia wartość w percepcji to stan chwytaka nad klockiem, tak? Warto by więc `'0'` zastąpić stałą np. `BLOCK_UNDER_GRIPPER = '0'` i zrobić `self.observation.append(BLOCK_UNDER_GRIPPER)`